src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.cs

```diff
@@ -451,7 +485,15 @@ private SyntaxTree WithChanges(SourceText newText, IReadOnlyList<TextChangeRange
             using (var parser = new InternalSyntax.LanguageParser(lexer, oldTree?.GetRoot(), changes))
             {
                 var compilationUnit = (CompilationUnitSyntax)parser.ParseCompilationUnit().CreateRed();
-                var tree = new ParsedSyntaxTree(newText, newText.Encoding, newText.ChecksumAlgorithm, this.FilePath, this.Options, compilationUnit, parser.Directives);
+                var tree = new ParsedSyntaxTree(
+                    newText,
+                    newText.Encoding,
+                    newText.ChecksumAlgorithm,
+                    this.FilePath,
+                    this.Options,
+                    compilationUnit,
+                    parser.Directives,
+                    DiagnosticOptions);
```

Comment:
>DiagnosticOptions [](start = 20, length = 17)

The other options from `this` tree are passed via `this.`. Please either correct the other options to be simple <property name> or use `this.DiagnosticOptions`. As written it makes it look that this value is from a different source than the `this` options above